Add recipe for 'terraform-cli'.

pataquets opened the pull request:
### Brief summary of what the package does

Provides an Emacs interface to Terraform CLI. This package is heavily inspired
on the awesome [Magit](https://magit.vc) package and uses [Transient](https://github.com/magit/transient) for its UI.

### Direct link to the package repository

https://github.com/pataquets/terraform-cli.el

### Your association with the package

I'm the author/maintainer. First published package. :sweat_smile: 

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
```
7:22: warning: Use a properly versioned dependency on "transient" if possible.
```
No specific version is required
```
85:0: error: `terraform-cli-plan:-refresh' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
```
There are other `transient:terraform-cli:-*` functions. Intentionally keeping name for grouping.
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`terraform-cli.el` with byte-compile using Emacs 28.1:
```
terraform-cli.el:38:31: Warning: the function `string-join' is not known to be
    defined.
terraform-cli.el:18:8: Warning: the function `package-installed-p' is not
    known to be defined.
```
`string-join` is an Emacs 29 function I believe (but you depend on Emacs 24.4).  `package-installed-p` probably needs a `(require 'package)`.

`terraform-cli.el` with checkdoc 0.6.2 (fix *within reason*):
```
terraform-cli.el:93: All variables and subroutines might as well have a documentation string
terraform-cli.el:96: All interactive functions should have documentation
terraform-cli.el:117: All variables and subroutines might as well have a documentation string
terraform-cli.el:120: All interactive functions should have documentation
```

`terraform-cli.el` with package-lint 20230905.629:
```
1 issue found:
7:22: warning: Use a properly versioned dependency on "transient" if possible.
```

`terraform-cli.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- terraform-cli.el#L40: No `format` required; `message` takes an f-string
```

- terraform-cli.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

pataquets commented:
Thanks for the prompt review, @riscy!
* Added SPDX license header, removed unneeded `format` (see linking commits).
* `string-join`appeared first on [24.4](https://git.savannah.gnu.org/cgit/emacs.git/tree/etc/NEWS.24#n1296), as linter already complained to me (`49:30: error: You should depend on (emacs "24.4") if you need 'string-join'.`).
* Checkdoc complaints are for not-meant-to-be-called functions. Left intentionally _within reason_ (unless they're mandatory).
* Also, my `transient` dependency usage is not requiring a specific version. Furthermore, I don't want to risk by requiring a "too high" version, without knowledge (currently limited).
* I'm not getting any error for `package-installed-p`, but I guess it might be my setup. I could use some help when choosing between inserting the `(require 'package)` you pointed or wrapping everything in `(with-eval-after-load 'package)`. Any tips?

Thanks in advance.

riscy commented:
> string-join

Your file needs to `(require 'subr-x)`; not every Emacs runtime will have brought this in at compile-time

> package-installed-p

`with-eval-after-load` will trigger a package-lint warning but this line - `(when (package-installed-p 'transient-showcase) (require 'transient-showcase))` - if you want to "optionally" bring this dependency in, use e.g. `(require 'transient-showcase nil t)` to avoid triggering errors if the package isn't installed.

> my transient dependency usage is not requiring a specific version

Per package-lint, you should do this if it's possible.  It doesn't really matter in practice (MELPA for example doesn't keep source for earlier versions of packages) but it lets hypothetical people who are trying to figure out bugs know *a* version of this dependency that worked in the past.  That can be very helpful.

riscy commented:
I'll mark this awaiting-upstream for now, but feel free to reach out!

riscy commented:
Friendly ping :)

pataquets commented:
Thanks, @riscy! Somehow I stopped working with the package, but I've resumed using it again and (and more heavily). And as my Elisp-fu improves, I've started thinking about picking this up again, since I think I can handle it better. 
Actually, I was talking this week about this stalled issue with a colleague 😁.

riscy commented:
Thank you! Let me know when this is ready for another look.

tarsius commented:
I am closing this and other pull-requests from 2023 because it is 2025 now and there hasn't been any recent activity.

In practical terms that has two consequences. The pull-request no longer appears on our todo list, and once/if you inform us that the package is ready for another round of reviews, we have to click on `Reopen pull-request` first.

I.e., it does not mean that you package has been rejected, just that we have reason to believe you might not follow up.
